src/main/kotlin/app/revanced/patches/youtube/layout/hide/getpremium/bytecode/fingerprints/GetPremiumVideoAdvertisementViewFingerprint.kt

```diff
@@ -0,0 +1,17 @@
+package app.revanced.patches.youtube.layout.hide.getpremium.bytecode.fingerprints
+
+import app.revanced.patcher.fingerprint.method.impl.MethodFingerprint
+import org.jf.dexlib2.Opcode
+
+object GetPremiumVideoAdvertisementViewFingerprint : MethodFingerprint(
+    opcodes = listOf(
+        Opcode.ADD_INT_2ADDR,
+        Opcode.ADD_INT_2ADDR,
+        Opcode.INVOKE_VIRTUAL,
+        Opcode.RETURN_VOID
+    ),
+    customFingerprint = { methodDef ->
+        methodDef.definingClass == "Lcom/google/android/apps/youtube/app/red/presenter/CompactYpcOfferModuleView;"
+                && methodDef.name == "onMeasure"
+    }
+)
```

Comment:
This fingerprint suggests that this patch will work with any version of the app; for that reason, you might like to remove the constraint on versions.

Reply:
It is very likely not gonna break in the future assuming the class and method stays, but if you think it might, then the constraint can remain.